src/ui/public/filter_editor/lib/filter_editor_utils.js

```diff
@@ -81,3 +81,10 @@ export function buildFilter({ indexPattern, field, operator, params, filterBuild
   filter.meta.negate = operator.negate;
   return filter;
 }
+
+export function areIndexPatternsProvided(indexPatterns) {
+  if (indexPatterns === undefined || indexPatterns.length === 0 || indexPatterns[0] === undefined) {
```

Comment:
What about just `if (_.compact(indexPatterns).length === 0)`? This will catch all of these conditions and even other possibly dangerous ones like `[undefined, null]`, etc.

Reply:
Great idea. I have updated the check with _.compact